probe_src/probe_py/cli.py

```diff
@@ -116,4 +116,86 @@ def dump(
                     print(op.data)
                 print()
 
+@app.command(
+    context_settings=dict(
+        ignore_unknown_options=True,
+    ),
+)
+def ssh(
+        ssh_args: list[str],
+        debug: bool = typer.Option(default=False, help="Run verbose & debug build of libprobe"),
+):
+    """
+    Wrap SSH and record provenance of the remote command.
+    """
+
+    if "-i" not in ssh_args:
+        typer.secho("RSA key not defined", fg=typer.colors.RED)
+        raise typer.Abort()
+
+    rsa_index = ssh_args.index("-i")
+    rsa = ssh_args[rsa_index + 1]
+
+    # Determine the correct library to use
+
+    libprobe = project_root / "libprobe/build" / ("libprobe-dbg.so" if debug else "libprobe.so")
+    if not libprobe.exists():
+        typer.secho(f"Libprobe not found at {libprobe}", fg=typer.colors.RED)
+        raise typer.Abort()
+
+    # Create a temporary directory on the local machine
+    local_temp_dir = pathlib.Path(tempfile.mkdtemp(prefix=f"probe_log_{os.getpid()}"))
+
+    # Check if remote platform matches local platform
+    remote_gcc_machine_cmd = ["ssh"] + ssh_args + ["gcc", "-dumpmachine"]
```

Comment:
`ssh_args` may contain a command we have to strip off, e.g., `ssh user@remote ls -l path`. We would want to strip off `ls`. In order to determine the point at which the SSH options end and the remote command begins, notice that the SSH options all either begin with a dash or are paired with options that begin with a dash, e.g. `-i path/to/key`. Therefore, the first argument which is neither must be the `remote` (or `user@remote`, but it might not have an `@`). The SSH options include all of the arguments up-to-and-including this first non-dashed and non-paired argument.